functions/go/starlark/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+)
+
+const (
+	helpText = `Run a starlark script to update or validate resources.
+
+There are 3 ways to specify your starlark source:
+1) Using inline source
+2) Using a file path
+3) Using a URL
+
+To specify the starlark source inline, you can use the configuration like this:
+
+apiVersion: kpt.dev/v1beta1
+kind: StarlarkFunction
+metadata:
+  name: my-star-fn
+source:
+  inline: |
+    # set the namespace on each resource
+    def run(r, ns_value):
+      for resource in r:
+        # mutate the resource
+        resource["metadata"]["namespace"] = ns_value
+    # get the value to add
+    ns_value = ctx.resource_list["functionConfig"]["keyValues"]["foo"]
+    run(ctx.resource_list["items"], ns_value)
+keyValues:
```

Comment:
`data` for brevity? Is this used already and would break things?

Reply:
> Is this used already and would break things?

No. 

I will change it to `data`.

Reply:
Actually, what's the use case for this?

In this example, isn't it easier to just inline `baz` instead of referring to `ctx.resource_list["functionConfig"]["keyValues"]["foo"]`

Reply:
These key-value pairs can be considered as parameters to the starlark script.
The user journey: A user writes a starlark script. The user wants to use this script in different pipelines to achieve slightly different customizations without changing the starlark script. The user can provide different values in these key-value pairs to achieve that.

Reply:
It sounds like you want reusability of scripts which is at odds with how inline works. One way to address this is to have two KRM resources:

StarlarkInvocation: This can either inline the script or have a reference to the StarlarkScript resource with key/value params.
StarlarkScript: Reusable script.

Maybe start with just StarlarkInvocation with inline script (not separate key/value). Later on you can introduce StarlarkScript.

Meta point: This kind of design should be done as a brief doc.

Reply:
> start with just StarlarkInvocation with inline script (no separate key/value field)

It seems ConfigMap can do what we want here.

Reply:
We shouldn't abuse ConfigMap unless we really need simple key/value pairs. In this case, we need a custom resource which can be extended in the future when you introduce `StarlarkScript`. I wouldn't recommend using ConfigMap here. `StarlarkRun` may be a better name than `StarlarkInvocation`.